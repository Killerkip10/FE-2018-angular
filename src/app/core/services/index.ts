export * from './auth.service';
export * from './profile.service';
export * from './token.service';
export * from './search-user.service';
