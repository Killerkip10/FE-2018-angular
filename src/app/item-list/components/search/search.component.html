<mat-form-field>
  <input #searchItem matInput placeholder="Search">
</mat-form-field>
