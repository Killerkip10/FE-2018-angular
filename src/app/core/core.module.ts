import {NgModule} from '@angular/core';
import {BrowserModule} from '@angular/platform-browser';
import {TranslateModule} from '@ngx-translate/core';
import {HttpClientModule} from '@angular/common/http';
import {RouterModule} from '@angular/router';

import {BrowserAnimationsModule} from '@angular/platform-browser/animations';
import {AuthGuard} from './guards';
import {AuthService} from './services';
import {
  MatButtonModule,
  MatToolbarModule,
  MatSelectModule,
  MatIconModule
} from '@angular/material';
import {
  HeaderComponent,
  FooterComponent
} from './components';

@NgModule({
  imports: [
    BrowserModule,
    HttpClientModule,
    TranslateModule,
    BrowserAnimationsModule,

    RouterModule,

    MatIconModule,
    MatButtonModule,
    MatSelectModule,
    MatToolbarModule
  ],
  declarations: [
    HeaderComponent,
    FooterComponent
  ],
  providers: [
    AuthService,

    AuthGuard
  ],
  exports: [
    HeaderComponent,
    FooterComponent
  ]
})
export class CoreModule {}
