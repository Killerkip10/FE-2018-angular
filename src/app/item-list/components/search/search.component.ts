import {ChangeDetectionStrategy, Component, ElementRef, EventEmitter, OnInit, Output, ViewChild} from '@angular/core';
import {fromEvent} from 'rxjs';
import {debounceTime} from 'rxjs/operators';

interface InputEvent {
  target: {value: string};
}

@Component({
  selector: 'app-search',
  templateUrl: './search.component.html',
  styleUrls: ['./search.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class SearchComponent implements OnInit {
  @ViewChild('searchItem') searchItem: ElementRef;
  @Output() search = new EventEmitter<string>();

  constructor() { }

  ngOnInit() {
    fromEvent<InputEvent>(this.searchItem.nativeElement, 'input')
      .pipe(debounceTime(500))
      .subscribe(v => this.search.emit(v.target.value));
  }
}
