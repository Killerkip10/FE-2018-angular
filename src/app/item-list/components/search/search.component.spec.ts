import {async, ComponentFixture, TestBed} from '@angular/core/testing';
import {ChangeDetectionStrategy, NO_ERRORS_SCHEMA} from '@angular/core';

import {SearchComponent} from './search.component';

describe('SearchComponent', () => {
  let component: SearchComponent;
  let fixture: ComponentFixture<SearchComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        SearchComponent
      ],
      schemas: [
        NO_ERRORS_SCHEMA
      ]
    })
      .overrideComponent(SearchComponent, {set: {changeDetection: ChangeDetectionStrategy.Default}})
      .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(SearchComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
