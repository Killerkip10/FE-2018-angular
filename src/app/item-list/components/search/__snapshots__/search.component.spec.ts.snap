// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchComponent snapshot 1`] = `
<app-search
  search={[Function EventEmitter]}
  searchItem={[Function ElementRef]}
>
  <mat-form-field>
    <input
      matinput=""
      placeholder="Search"
    />
  </mat-form-field>
</app-search>
`;
